services/provider/server/storagerequest.go

```diff
@@ -32,45 +32,51 @@ func newStorageRequestManager(cl client.Client, namespace string) (*storageReque
 	}, nil
 }
 
-// getStorageRequestName generates a name for a StorageRequest resource.
-func getStorageRequestName(consumerUUID, storageRequestName string) string {
-	var s struct {
+// getStorageRequestHash generates a hash for a StorageRequest based
+// on the MD5 hash of the StorageClaim name and storageConsumer UUID.
+func getStorageRequestHash(consumerUUID, storageClaimName string) string {
+	s := struct {
 		StorageConsumerUUID string `json:"storageConsumerUUID"`
-		StorageRequestName  string `json:"storageRequestName"`
+		// name in json tag isn't changed for backward compatibility or else md5 sum of json marshal will change
+		StorageClaimName string `json:"storageRequestName"`
```

Comment:
The secret names are already changing so why do we care about changing this?

Reply:
This isn't for secret name but for the name of the storagerequest itself. Earlier it was `storageclassrequest-hash123` now it'll be `storagerquest-hash123` it needs to be like this for migration.

Reply:
changed the tag as well, done. Since underlying ceph resources names itself will be changed, having this backward compatible doesn't server any purpose, will raise new PR for a consistent naming of cbprns and cfssvg.